compiler/src/main/kotlin/visitors/grapher.kt

```diff
@@ -0,0 +1,305 @@
+package dk.aau.cs.d409f19.cellumata.visitors
+
+import dk.aau.cs.d409f19.cellumata.ast.*
+import java.io.OutputStream
+import java.io.PrintStream
+
+class ASTGrapher(sink: OutputStream, private val output: PrintStream = PrintStream(sink)) : BaseASTVisitor() {
+    private var depth = 0
+
+    private fun printNode(from: Any, to: Any) {
+        output.println("\t" + from.hashCode() + " -> " + to.hashCode())
+    }
+
+    private fun printLabel(node: Any, label: String) {
+        output.println("\t" + node.hashCode() + " [label=\"" + label + "\"]")
+    }
+
+    override fun visit(node: WorldNode) {
+        var builder = StringBuilder()
+            .append("WorldNode\\nsize = ")
+            .append(node.dimensions.map {
+                when(it.type) {
+                    WorldType.EDGE -> it.size.toString() + "[edge=" + it.edge + "]"
+                    WorldType.UNDEFINED -> it.size.toString() + "[undefined]"
+                    WorldType.WRAPPING -> it.size.toString() + "[wrapping]"
+                }
+            }.joinToString(", "))
+
+        if (node.cellSize != null) {
+            builder = builder.append("\\ncellsize=").append(node.cellSize!!)
+        }
+
+        if (node.tickrate != null) {
+            builder = builder.append("\\ntickrate=").append(node.tickrate!!)
+        }
+
+        printLabel(node, builder.toString())
+        super.visit(node)
+    }
+
+    override fun visit(node: RootNode) {
```

Comment:
Could we at least have the RootNode fun override be on top? I don't ask for a completely ordered hierarchy, but some structure could be nice.

Reply:
This seem unimportant

Reply:
Well yes, but actually no. It's important for readability.